analysers/analyser_osmosis_highway_area_access.py

```diff
@@ -51,24 +51,120 @@
   )
 """
 
+sql20 = """
+CREATE TEMP TABLE {barriertype} AS
+SELECT
+  barrier.id AS id,
+  barrier.tags AS tags,
+  ways.id as wid,
+  ST_AsText(barrier.geom) AS geom
+FROM
+  nodes AS barrier
+  JOIN way_nodes ON
+    way_nodes.node_id = barrier.id
+  JOIN highways AS ways ON
+    ways.id = way_nodes.way_id AND
+    NOT ways.is_construction AND
+    NOT ways.is_area
+  LEFT JOIN highway_ends ON
+    ways.id = highway_ends.id AND
+    barrier.id = highway_ends.nid
```

Comment:
There is faster way to test if the node is at the ends that not require a left join. Just look into the nodes array.